src/main/java/com/github/manolo8/darkbot/core/objects/RefinementGui.java

```diff
@@ -71,12 +74,17 @@ public void update() {
 
         @Override
         public void update(long address) {
-            if (address != this.address || name == null || !name.contains("ore")) {
+            if (address != this.address || name == null || (!name.contains("ore") && !name.contains("lab"))) {
                 name = API.readMemoryString(address, 184);
 
                 if (name != null && !name.isEmpty()) {
-                    String processedName = name.replace("ore_", "");
-                    fuzzyName = processedName.substring(0, 1).toUpperCase(Locale.ROOT) + processedName.substring(1);
+                    if(name.contains("ore_")) {
+                        String processedName = name.replace("ore_", "");
+                        fuzzyName = processedName.substring(0, 1).toUpperCase(Locale.ROOT) + processedName.substring(1);
+                    }
+                    if(name.contains("lab_")){
+                        fuzzyName = API.readMemoryString(address, 0x108);
```

Comment:
the previous one gets the fuzzyName out of name, but this one doesn't
is there a reason for it? 
if so, could you leave it as a comment?
otherwise, would this work if the logic was the same for both?
Maybe document what kind of strings these are as part of a comment

Also watch the formatting `if (...) {`, with spaces between if and before {

Reply:
You can also use `if {} else if {}` so that it will not do the second check if it has already entered the first `if`